### ISSUE TITLE
Excluding rspec shared examples from generated documentation

### ISSUE THREAD

**Issue by damien** [zipmark/rspec_api_documentation#47]
Hey all!

First off, thanks so much for this gem! It's saving me a lot of time working with consumers of an API I'm prototyping!

That being said, I've currently having trouble configuring rspec_api_documentation; specifically the `filter` and `exclude_filter` options in [`RspecApiDocumentation ::Configuration`](https://github.com/zipmark/rspec_api_documentation/commit/7db5d1cdad9d2abd8dd651ce2f28feab61df9042#lib/rspec_api_documentation/configuration.rb).

My use case is as follows: I have a set of shared examples that look something like this:

``` ruby
shared_examples "an api endpoint", standard_response: true do
  example_request "should respond with HTTP 200" do
    status.should == 200
    response_body.should =~ /"status": 200/
  end

  example_request "should return a response with a json mime type" do
    response_headers["Content-Type"].should == "application/json; charset=utf-8"
  end
end
```

It's useful to me to have these run for a variety of examples. Unfortunately, these examples are not at all helpful when they're included in the generated documentation. I've attempted to exclude them by adding `document: false` and later `document: :private` with the corresponding exclude filter configuration. Neither has worked out too well for me.

**TLDR: How do I exclude shared examples from the generated documentation (rake docs:generate)?**

Thanks for your time, and awesome work with this gem!
